### ISSUE TITLE
Show an alert/what's new message when a new feature is introduced

### ISSUE THREAD

**Issue by agrogers** [AndBible/and-bible#2448]
I reckon most of our users don't use some feature because they don't know it exists or what it does. Could we add an option to show a simple text message in a nice dialog, perhaps with the ability to step through additional messages?

Worthy candidates would be:
1. Workspace swipe options
2. New major release being pushed to the play store.
3. Swipe options to minimise and maximise
4. Changing workspace colors

Releated to #2353 and #1239

**Comment by tuomas2** [comment]
I don't like the idea of spamming all users about features that only minority of users will need. When a new (major) version is released, an introductory video of all (major) new features is proper, of course.

**Comment by agrogers** [comment]
Users can't discover features they need if they never know they are there.
And I dont think a high percentage of users go to the videos. I do think
real value would be added for the majority if AB had an onboarding
experience for new users and alerts for existing users of new UI features
(and even old features that many dont know about).

Adding a button to the dialog that says 'Only show major features' would
avoid the spam.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tuomas2** [comment]
We advertise the "new features" video when new version is published. It's on users responsibility to watch or not watch it. I don't want to force users to go through anything extra introductory material. And introductory material should be translated... (or perhaps this is something that could be kept english only and not displayed for non-english speakers). 

Anyway any introductory material should be 100% optional (meaning, there should be a button to skip everything and go back to the business).

**Comment by agrogers** [comment]
Yeah, agreed on 100% optional.

Having gone through some of the PlayStore comments today, I can see a few
that may have given us a better rating if we had had an onboarding / tips
for them to read because they couldn't work out how to do simple things. Of
course they could just skip that too.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by agrogers** [comment]
This is an example (on a web app) that makes me feel more that we should alert users to new features. It looks great, takes very little time to dismiss and tells me I can do something I had no idea about.

[By the way, Quire is really good and the free plan feature rich]

![image](https://user-images.githubusercontent.com/13920678/213099245-f857bd0a-4ced-468d-bcad-fc1bb9e1db6e.png)

**Comment by tuomas2** [comment]
To be honest, I don't remember ever reading those, just finding quickly way to close the sudden annoyance :-P So (though maybe I have already said it, saying again that) I'm not that into them. I would stick with providing a way to figure things out, not force-pushing education to user. So when a new version is released it contains link to a video tutorials and could also have a button "show me how to use new features" etc, which then opens such a tutorial. But not anything more forceful I think.

**Comment by agrogers** [comment]
I think you underestimate your intelligence or overestimate everyone else's
:)  When it comes to apps and digital intelligence there are novice users,
advanced users, power users... and then there is you. Regarding this issue,
I'm not thinking for you ;-)

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by agrogers** [comment]
I asked ChatGPT if there is a concensus about onboarding experiences or tips for Android apps. There isn't as of 2022. It did however give some considerations:

1. Complexity of the App: If your app has a complex user interface or offers advanced features, providing an onboarding experience can help users understand how to navigate and use the app effectively. _AB is defintiely more complex than all Bible apps I have used except MyBible._

2. Target Audience: If your app caters to a broad audience and onboarding experiences might be beneficial. For tech-savvy users, a more minimalist approach might be preferred. _I think the lack of extra "hand-holding" is one reason the audience isnt broader._

3. Nature of Features: If your app introduces new or unique features that users might not be familiar with, providing tips or guiding them through the functionalities can enhance the user experience. _Definitely yes to this._

4. Balancing Act: Strive for a balance between guiding users and not overwhelming them. _This is so easy to do. A simple check box that says "Never show me any tips or the onboarding experience again" will eliminate the annoyance for people who hate tips. For everyone else who likes to know what the app they use can do, it will be well received. And if they get sick of helpful tips, they can just tick the box._ 

I can see only miniscule downsides for just a few people if opt-out tips were added. And I see significant help for many people (including me) if they are added. I am sure there is a reason why so many big-name app providers include this sort of thing in their apps.

But we can agree to disagree on this one :)
